Build error when compiling for HTML5

**Godot version:** Git https://github.com/godotengine/godot/commit/a1033aea51515cd5f831cedf9834ea376200356a

**OS/device including version:** Fedora 30, Emscripten 1.38.48

**Issue description:** Building for HTML5 fails with the following error:

```text
Compiling ==> modules/websocket/emws_client.cpp
In file included from modules/websocket/emws_client.cpp:33:
modules/websocket/emws_client.h:53:8: warning: 'EMWSClient::connect_to_host' hides overloaded virtual function [-Woverloaded-virtual]
        Error connect_to_host(String p_host, String p_path, uint16_t p_port, bool p_ssl, const PoolVector<String> p_protocol = PoolVector<String>(), const Dictionary p_custom_headers = Dictionary());
              ^
modules/websocket/websocket_client.h:60:16: note: hidden overloaded virtual function 'WebSocketClient::connect_to_host' declared here: type mismatch at 5th parameter ('const Vector<String>' vs 'const PoolVector<String>')
        virtual Error connect_to_host(String p_host, String p_path, uint16_t p_port, bool p_ssl, const Vector<String> p_protocol = Vector<String>(), const Vector<String> p_custom_headers = Vector<String>()) = 0;
                      ^
In file included from modules/websocket/emws_client.cpp:33:
modules/websocket/emws_client.h:42:11: error: allocating an object of abstract class type 'EMWSClient'
        GDCIIMPL(EMWSClient, WebSocketClient);
                 ^
modules/websocket/websocket_client.h:60:16: note: unimplemented pure virtual method 'connect_to_host' in 'EMWSClient'
        virtual Error connect_to_host(String p_host, String p_path, uint16_t p_port, bool p_ssl, const Vector<String> p_protocol = Vector<String>(), const Vector<String> p_custom_headers = Vector<String>()) = 0;
                      ^
modules/websocket/emws_client.cpp:67:19: error: out-of-line definition of 'connect_to_host' does not match any declaration in 'EMWSClient'
Error EMWSClient::connect_to_host(String p_host, String p_path, uint16_t p_port, bool p_ssl, const PoolVector<String> p_protocols, const Vector<String> p_custom_headers) {
                  ^~~~~~~~~~~~~~~
modules/websocket/emws_client.cpp:69:24: error: 'this' argument to member function 'join' has type 'const PoolVector<String>', but function is not marked const
        String proto_string = p_protocols.join(",");
                              ^~~~~~~~~~~
./core/pool_vector.h:443:9: note: 'join' declared here
        String join(String delimiter) {
               ^
1 warning and 3 errors generated.
shared:ERROR: '/builds/Calinou/godot-builds-ci/godot/emsdk/fastcomp/fastcomp/bin/clang++ -target asmjs-unknown-emscripten -D__EMSCRIPTEN_major__=1 -D__EMSCRIPTEN_minor__=38 -D__EMSCRIPTEN_tiny__=48 -D_LIBCPP_ABI_VERSION=2 -Werror=implicit-function-declaration -Xclang -nostdsysteminc -Xclang -isystem/builds/Calinou/godot-builds-ci/godot/emsdk/fastcomp/emscripten/system/include/libcxx -Xclang -isystem/builds/Calinou/godot-builds-ci/godot/emsdk/fastcomp/emscripten/system/lib/libcxxabi/include -Xclang -isystem/builds/Calinou/godot-builds-ci/godot/emsdk/fastcomp/emscripten/system/include/compat -Xclang -isystem/builds/Calinou/godot-builds-ci/godot/emsdk/fastcomp/emscripten/system/include -Xclang -isystem/builds/Calinou/godot-builds-ci/godot/emsdk/fastcomp/emscripten/system/include/libc -Xclang -isystem/builds/Calinou/godot-builds-ci/godot/emsdk/fastcomp/emscripten/system/lib/libc/musl/arch/emscripten -Xclang -isystem/builds/Calinou/godot-builds-ci/godot/emsdk/fastcomp/emscripten/system/local/include -c -std=c++11 -Os -fno-exceptions -fno-rtti -Wall -Werror=return-type -DJAVASCRIPT_ENABLED -DUNIX_ENABLED -DNO_THREADS -DNO_SAFE_CAST -DJAVASCRIPT_EVAL_ENABLED -DNDEBUG -DPTRCALL_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -DMODULE_BMP_ENABLED -DMODULE_BULLET_ENABLED -DMODULE_CSG_ENABLED -DMODULE_DDS_ENABLED -DMODULE_ENET_ENABLED -DMODULE_FREETYPE_ENABLED -DMODULE_GDNATIVE_ENABLED -DMODULE_GDSCRIPT_ENABLED -DMODULE_GRIDMAP_ENABLED -DMODULE_HDR_ENABLED -DMODULE_JPG_ENABLED -DMODULE_JSONRPC_ENABLED -DMODULE_MOBILE_VR_ENABLED -DMODULE_OGG_ENABLED -DMODULE_OPENSIMPLEX_ENABLED -DMODULE_OPUS_ENABLED -DMODULE_PVR_ENABLED -DMODULE_REGEX_ENABLED -DMODULE_SQUISH_ENABLED -DMODULE_STB_VORBIS_ENABLED -DMODULE_SVG_ENABLED -DMODULE_TGA_ENABLED -DMODULE_THEORA_ENABLED -DMODULE_UPNP_ENABLED -DMODULE_VHACD_ENABLED -DMODULE_VISUAL_SCRIPT_ENABLED -DMODULE_VORBIS_ENABLED -DMODULE_WEBM_ENABLED -DMODULE_WEBP_ENABLED -DMODULE_WEBRTC_ENABLED -DMODULE_WEBSOCKET_ENABLED -Ithirdparty/libpng -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/javascript -I. -DEMSCRIPTEN -mllvm -disable-llvm-optzns modules/websocket/emws_client.cpp -fno-inline-functions -Xclang -disable-O0-optnone -Xclang -isystem/builds/Calinou/godot-builds-ci/godot/emsdk/fastcomp/emscripten/system/include/SDL -c -o modules/websocket/emws_client.javascript.opt.bc -emit-llvm' failed (1)
scons: *** [modules/websocket/emws_client.javascript.opt.bc] Error 1
scons: building terminated because of errors.
```

[Full build log on GitLab CI](https://gitlab.com/Calinou/godot-builds-ci/-/jobs/322033932)

**Steps to reproduce:** Build Godot using `scons platform=javascript tools=no target=release`.

--- a/modules/websocket/emws_client.cpp
+++ b/modules/websocket/emws_client.cpp
@@ -64,9 +64,15 @@ EMSCRIPTEN_KEEPALIVE void _esws_on_close(void *obj, int code, char *reason, int 
 }
 }
 
-Error EMWSClient::connect_to_host(String p_host, String p_path, uint16_t p_port, bool p_ssl, const PoolVector<String> p_protocols, const Vector<String> p_custom_headers) {
+Error EMWSClient::connect_to_host(String p_host, String p_path, uint16_t p_port, bool p_ssl, const Vector<String> p_protocols, const Vector<String> p_custom_headers) {
 
-	String proto_string = p_protocols.join(",");
+	String proto_string;
+	for (int i = 0; i < p_protocols.size(); i++) {
+		if (i != 0)
+			proto_string += ",";
+		proto_string += p_protocols[i];
+	}
+
 	String str = "ws://";
 
 	if (p_custom_headers.size()) {

--- a/modules/websocket/emws_client.h
+++ b/modules/websocket/emws_client.h
@@ -50,7 +50,7 @@ public:
 	bool _is_connecting;
 
 	Error set_buffers(int p_in_buffer, int p_in_packets, int p_out_buffer, int p_out_packets);
-	Error connect_to_host(String p_host, String p_path, uint16_t p_port, bool p_ssl, const PoolVector<String> p_protocol = PoolVector<String>(), const Dictionary p_custom_headers = Dictionary());
+	Error connect_to_host(String p_host, String p_path, uint16_t p_port, bool p_ssl, const Vector<String> p_protocol = Vector<String>(), const Vector<String> p_custom_headers = Vector<String>());
 	Ref<WebSocketPeer> get_peer(int p_peer_id) const;
 	void disconnect_from_host(int p_code = 1000, String p_reason = "");
 	IP_Address get_connected_host() const;

--- a/modules/websocket/emws_server.cpp
+++ b/modules/websocket/emws_server.cpp
@@ -33,7 +33,7 @@
 #include "emws_server.h"
 #include "core/os/os.h"
 
-Error EMWSServer::listen(int p_port, PoolVector<String> p_protocols, bool gd_mp_api) {
+Error EMWSServer::listen(int p_port, Vector<String> p_protocols, bool gd_mp_api) {
 
 	return FAILED;
 }

--- a/modules/websocket/emws_server.h
+++ b/modules/websocket/emws_server.h
@@ -43,7 +43,7 @@ class EMWSServer : public WebSocketServer {
 
 public:
 	Error set_buffers(int p_in_buffer, int p_in_packets, int p_out_buffer, int p_out_packets);
-	Error listen(int p_port, PoolVector<String> p_protocols = PoolVector<String>(), bool gd_mp_api = false);
+	Error listen(int p_port, Vector<String> p_protocols = Vector<String>(), bool gd_mp_api = false);
 	void stop();
 	bool is_listening() const;
 	bool has_peer(int p_id) const;

--- a/modules/websocket/wsl_client.cpp
+++ b/modules/websocket/wsl_client.cpp
@@ -181,8 +181,12 @@ Error WSLClient::connect_to_host(String p_host, String p_path, uint16_t p_port, 
 	_connection = _tcp;
 	_use_ssl = p_ssl;
 	_host = p_host;
-	_protocols.clear();
-	_protocols.append_array(p_protocols);
+	// Strip edges from protocols.
+	_protocols.resize(p_protocols.size());
+	String *pw = _protocols.ptrw();
+	for (int i = 0; i < p_protocols.size(); i++) {
+		pw[i] = p_protocols[i].strip_edges();
+	}
 
 	_key = WSLPeer::generate_key();
 	// TODO custom extra headers (allow overriding this too?)

--- a/modules/websocket/wsl_server.cpp
+++ b/modules/websocket/wsl_server.cpp
@@ -80,11 +80,12 @@ bool WSLServer::PendingPeer::_parse_request(const Vector<String> p_protocols) {
 	if (headers.has("sec-websocket-protocol")) {
 		Vector<String> protos = headers["sec-websocket-protocol"].split(",");
 		for (int i = 0; i < protos.size(); i++) {
+			String proto = protos[i].strip_edges();
 			// Check if we have the given protocol
 			for (int j = 0; j < p_protocols.size(); j++) {
-				if (protos[i] != p_protocols[j])
+				if (proto != p_protocols[j])
 					continue;
-				protocol = protos[i];
+				protocol = proto;
 				break;
 			}
 			// Found a protocol
@@ -158,7 +159,12 @@ Error WSLServer::listen(int p_port, const Vector<String> p_protocols, bool gd_mp
 	ERR_FAIL_COND_V(is_listening(), ERR_ALREADY_IN_USE);
 
 	_is_multiplayer = gd_mp_api;
-	_protocols.append_array(p_protocols);
+	// Strip edges from protocols.
+	_protocols.resize(p_protocols.size());
+	String *pw = _protocols.ptrw();
+	for (int i = 0; i < p_protocols.size(); i++) {
+		pw[i] = p_protocols[i].strip_edges();
+	}
 	_server->listen(p_port);
 
 	return OK;

